src/main/java/shop/kkeujeok/kkeujeokbackend/challenge/application/util/ChallengeBlockStatusUpdateScheduler.java

```diff
@@ -0,0 +1,48 @@
+package shop.kkeujeok.kkeujeokbackend.challenge.application.util;
+
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+import shop.kkeujeok.kkeujeokbackend.block.domain.Block;
+import shop.kkeujeok.kkeujeokbackend.block.domain.Type;
+import shop.kkeujeok.kkeujeokbackend.block.domain.repository.BlockRepository;
+import shop.kkeujeok.kkeujeokbackend.global.entity.Status;
+import shop.kkeujeok.kkeujeokbackend.member.domain.Member;
+import shop.kkeujeok.kkeujeokbackend.notification.application.NotificationService;
+
+@Component
+@RequiredArgsConstructor
+public class ChallengeBlockStatusUpdateScheduler {
+
+    private final BlockRepository blockRepository;
+    private final NotificationService notificationService; // NotificationService 주입
+
+    @Scheduled(cron = "0 0 0 * * ?")
+    @Transactional
+    public void updateStatuses() {
+        List<Block> blocks = blockRepository.findByType(Type.CHALLENGE);
+
+        blocks.forEach(block -> {
+            Status previousStatus = block.getStatus();
+            Status newStatus;
+
+            if (!ChallengeBlockStatusUtil.isChallengeBlockActiveToday(block.getChallenge().getCycle(),
+                    block.getChallenge().getCycleDetails())) {
+                newStatus = Status.UN_ACTIVE;
+            } else {
+                newStatus = Status.ACTIVE;
+            }
```

Comment:
if 조건 내에 부정 표현보단 긍정문을 통해 가독성을 향상 시키는 방향은 어떠신가요??

```suggestion
            if (ChallengeBlockStatusUtil.isChallengeBlockActiveToday(block.getChallenge().getCycle(),
                    block.getChallenge().getCycleDetails())) {
                newStatus = Status.ACTIVE;
            } else {
                newStatus = Status.UN_ACTIVE;
            }
```

Reply:
확실히 if 조건 내에 부정문보다는 긍정문이 가독성이 더 좋은 것 같습니다. 수정하도록 하겠습니다!